src/v_draw.cpp

```diff
@@ -958,6 +958,20 @@ void DCanvas::DrawLine(int x0, int y0, int x1, int y1, int palColor, uint32_t re
 #endif
 }
 
+DEFINE_ACTION_FUNCTION(_Screen, DrawLine)
+{
+	PARAM_PROLOGUE;
+	PARAM_INT(x0);
+	PARAM_INT(y0);
+	PARAM_INT(x1);
+	PARAM_INT(y1);
+	PARAM_INT(color);
```

Comment:
One thing to note is that the current software renderer implementation does not clip the line to the actual viewport. This would have to be added here before making it public.